### PR TITLE
Documentation -- Handling Data section reorganization

### PR DESCRIPTION
**Pull request by simonsonjack**
My primary change was that I took the Data Format section and broke it up into subsections for each data type. Each subsection contains a small code snippet that is meant to give the user a clear example of how they could access and use the different data types in their own code. I'll need help to make sure the html won't break since I don't know html well. Sorry there are so many commits in this one PR!

### CODE REVIEW

File: 02 Algorithm Reference/03 Handling Data/07 QuoteBars.html [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    ## access the QuoteBar dictionary and then subset this for your desired Forex symbol
+    fxQuoteBars = data.QuoteBars
+    eurusd      = fxQuoteBars['EURUSD']
+    fxOpen      = eurusd.Open          ## Market Open FX Rate
+    fxClose     = eurusd.Close         ## Market Close FX Rate
+            
+</pre>
+</div>
+<p>
+    With a specific QuoteBar you can also access Bid and Ask Bars for the same security. These Bars provide information specific to the Bid and Ask side of Forex and futures secuirites, whereas QuoteBars use the mean of the Open, High, Low, and Close attributes of the Bid and Ask Bars to extrapolte values.
```
Comment (jaredbroad) [inline 02 Algorithm Reference/03 Handling Data/07 QuoteBars.html]: Lets clarify this with an example:

<p>
The quotebar.Open, High, Low and Close properties are the midpoint of the Bid-Ask spread at that moment of time. 
</p>

<p>
The quote bar also contains two sub-objects; <code>quotebar.Bid</code> and <code>quotebar.Ask</code>. These Bid and Ask bars have Open, High, Low and close properties. e.g. <code>quotebar.Bid.Open</code>.
</p>

File: 02 Algorithm Reference/03 Handling Data/08 Dividends.html [hunk 2]
```diff
@@ -0,0 +1,22 @@
+<meta name="tag" content="data event handlers"/>
+<meta name="tag" content="dividends event"/>
+
+<p>
+    $[Dividend,T:QuantConnect.Data.Market.Dividend] events are triggered on payment of a dividend. It provides the Distribution per share.
+</p>
+<div class="section-example-container">
+<pre class="python">
+    self.spy = self.AddEquity("SPY", Resolution.Hour)
+    self.spy.SetDataNormalizationMode(DataNormalizationMode.Raw)
```
Comment (jaredbroad) [inline 02 Algorithm Reference/03 Handling Data/08 Dividends.html]: Not sure if this is required.

File: 02 Algorithm Reference/03 Handling Data/08 Dividends.html [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    
+def OnData(self, data):
+    if not self.Portfolio.Invested:
+        self.Buy("SPY", 100)
+    
+    ## Condition to see if SPY is in the Dividend DataDictionary
+    if data.Dividends.ContainsKey("SPY"):
+        ## Log the dividend distribution
+        self.Log(str(data.Dividends["SPY"].Distribution))
```
Comment (jaredbroad) [inline 02 Algorithm Reference/03 Handling Data/08 Dividends.html]: 👍

File: 02 Algorithm Reference/03 Handling Data/09 Splits.html [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+        self.Buy("MSFT", 100)
+    
+    ## If MSFT had a split, print out information about it
+    if data.Splits.ContainsKey("MSFT"):
+        ## Log split information
+        spySplit = data.Splits['MSFT']
+        if spySplit.Type == 0:
+            self.Log('MSFT stock will split next trading day')
+        if spySplit.Type == 1:
+            self.Log('Split type: ' + str(spySplit.Type) + '; Split factor: ' + str(spySplit.SplitFactor) +
```
Comment (jaredbroad) [inline 02 Algorithm Reference/03 Handling Data/09 Splits.html]: beautiful!

File: 02 Algorithm Reference/03 Handling Data/11 Delistings [hunk 5]
```diff
@@ -0,0 +1,27 @@
+<meta name="tag" content="data event handlers"/>
+<meta name="tag" content="delistings event"/>
+
+<p>
+    $[Delisting,T:QuantConnect.Data.Market.Delisting] events provide notice an asset is no longer trading on the exchange.
```
Comment (jaredbroad) [inline 02 Algorithm Reference/03 Handling Data/11 Delistings]: "A delisting warning is issued the final trading day for a stock delisting event to give you algorithm time to gracefully exit out of positions before forced termination"

File: 02 Algorithm Reference/03 Handling Data/06 TradeBars.html [hunk 6]
```diff
@@ -0,0 +1,24 @@
+<meta name="tag" content="data event handlers"/>
+<meta name="tag" content="tradebars event"/>
+<p>
+    $[TradeBar,T:QuantConnect.Data.Market.TradeBar] provides Open, High, Low, Close and Volume properties for a given period of time. TradeBars are only supported for equities, options, and futures contracts (NOT Forex).
+</p>
+<div class="section-example-container">
+<pre class="python">
```
[the remark this answers is not in the document]
Comment (jaredbroad) [reply 02 Algorithm Reference/03 Handling Data/06 TradeBars.html]: ``

File: 02 Algorithm Reference/03 Handling Data/07 QuoteBars.html [hunk 7]
```diff
@@ -0,0 +1,42 @@
+<meta name="tag" content="data event handlers"/>
+<meta name="tag" content="quotebars event"/>
+
+<p>
+    $[QuoteBar,T:QuantConnect.Data.Market.QuoteBar] provides Open, High, Low, Close, Bid, Ask, LastBidSize and LastAskSize properties for a given period of time. The Bid and the Ask properties are $[Bar,T:QuantConnect.Data.Market.Bar] objects that contains Open, High, Low and Close. The QuoteBar Open, High, Low and Close properties values are the mean of the respective Bid and Ask property. QuoteBars are supported only for Forex securities and futures contracts.
+</p>
+<div class="section-example-container">
+<pre class="python">
+    self.AddForex('EURUSD', Resolution.Hour) # Subscribe to hourly QuoteBars
```
[the remark this answers is not in the document]
Comment (jaredbroad) [reply 02 Algorithm Reference/03 Handling Data/07 QuoteBars.html]: ``

File: 02 Algorithm Reference/03 Handling Data/06 TradeBars.html (hunk 6 above)
[the remark this answers is not in the document]
Comment (grntmhqo) [reply 02 Algorithm Reference/03 Handling Data/06 TradeBars.html]: what is the status.i see this was closed but don't see rather it was a success
```suggestion
<pre class="python">
```

File: 02 Algorithm Reference/03 Handling Data/05 Ticks.html [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+    $[Ticks,T:QuantConnect.Data.Market.Tick] provides LastPrice and Quantity properties for a given time. If it is a quote, it also contains non-zero BidPrice, BidSize, AskPrice, AskSize properties.
+    
+<div class="section-example-container">
+<pre class="python">
+    self.AddEquity("IBM", Resolution.Tick) ## Subscribe to tick-level IBM data
+        
+def OnData(self, data):
+     ## Ticks can get pushed together and arrive at the same time
+     ## and have the same EndTime in the data handler, so we use 
+     ## the [-1] to access to most recent tick that arrived
```
[the remark this answers is not in the document]
Comment (grntmhqo) [reply 02 Algorithm Reference/03 Handling Data/05 Ticks.html]:
```suggestion
     ## the [-1] to access to most recent tick that arrived
```

File: 02 Algorithm Reference/03 Handling Data/05 Ticks.html [hunk 9]
```diff
@@ -0,0 +1,22 @@
+<meta name="tag" content="data event handlers"/>
+<meta name="tag" content="tradebars event"/>
+<meta name="tag" content="ticks event"/>
+
+<p>
+    $[Ticks,T:QuantConnect.Data.Market.Tick] provides LastPrice and Quantity properties for a given time. If it is a quote, it also contains non-zero BidPrice, BidSize, AskPrice, AskSize properties.
```
[the remark this answers is not in the document]
Comment (grntmhqo) [reply 02 Algorithm Reference/03 Handling Data/05 Ticks.html]:
```suggestion
    $[Ticks,T:QuantConnect.Data.Market.Tick] provides LastPrice and Quantity properties for a given time. If it is a quote, it also contains non-zero BidPrice, BidSize, AskPrice, AskSize properties.
```

File: 02 Algorithm Reference/03 Handling Data/02 Time Slices.html [hunk 10]
```diff
@@ -66,51 +66,11 @@
     It contains all the data for a given moment of time. TradeBars and QuoteBars are symbol/string indexed dictionaries
     so you
     can easily access the data. Ticks is a list of ticks for that moment of time; indexed by the symbol.
-</p>
-<p>
     To make accessing the data easier the Slice object itself can also be indexed. e.g. <code>slice["IBM"]</code> will
     return a
-    <b>TradeBar</b> for IBM. <code>slice["EURUSD"]</code> will return a <b>QuoteBar</b> for EURUSD.
+    <b>TradeBar</b> for IBM. <code>slice["EURUSD"]</code> will return a <b>QuoteBar</b> for EURUSD. TradeBars are supported for equities,
+    options, and futures contracts; QuoteBars are supported for Forex and futures contracts.
```
[the remark this answers is not in the document]
Comment (grntmhqo) [reply 02 Algorithm Reference/03 Handling Data/02 Time Slices.html]:
```suggestion
    options, and futures contracts; QuoteBars are supported for Forex and futures contracts.
```

File: 02 Algorithm Reference/03 Handling Data/01 Introduction.html [hunk 11]
```diff
@@ -10,8 +10,8 @@
 </p>
 <p>
 	All data uses <code>DataDictionary</code> objects to group data by symbol and provide easy access to information. The plural of
-	the type denotes the collection of objects e.g. the <i>TradeBars</i> DataDictionary is made up of <i>TradeBar</i> objects. You
+	the type denotes the collection of objects (e.g., the <i>TradeBars</i> DataDictionary is made up of <i>TradeBar</i> objects; <i>QutoeBars</i> DataDictionary is made up of <i>QuoteBar</i> objects). You
 	can access individual data points in the dictionary through its string or symbol dictionary index. For example 
 	<code class="csharp fsharp">var ibmTradeBar = slice.Bars["IBM"]</code>
-	<code class="python">ibmBar = slice.Bars["IBM"]</code>.
+	<code class="python">ibmTradeBar = slice.Bars["IBM"]</code>.
```
[the remark this answers is not in the document]
Comment (grntmhqo) [reply 02 Algorithm Reference/03 Handling Data/01 Introduction.html]:
```suggestion
	<code class="python">ibmTradeBar = slice.Bars["IBM"]</code>.
```

File: 02 Algorithm Reference/03 Handling Data/08 Dividends.html (hunk 3 above)
Comment (grntmhqo) [reply 02 Algorithm Reference/03 Handling Data/08 Dividends.html]:
```suggestion
        self.Log(str(data.Dividends["SPY"].Distribution))
```

File: 02 Algorithm Reference/03 Handling Data/08 Dividends.html (hunk 2 above)
Comment (grntmhqo) [reply 02 Algorithm Reference/03 Handling Data/08 Dividends.html]:
```suggestion
    self.spy.SetDataNormalizationMode(DataNormalizationMode.Raw)
```

File: 02 Algorithm Reference/03 Handling Data/07 QuoteBars.html (hunk 1 above)
Comment (grntmhqo) [reply 02 Algorithm Reference/03 Handling Data/07 QuoteBars.html]:
```suggestion
    With a specific QuoteBar you can also access Bid and Ask Bars for the same security. These Bars provide information specific to the Bid and Ask side of Forex and futures secuirites, whereas QuoteBars use the mean of the Open, High, Low, and Close attributes of the Bid and Ask Bars to extrapolte values.
```

File: 02 Algorithm Reference/03 Handling Data/07 QuoteBars.html [hunk 12]
```diff
@@ -0,0 +1,42 @@
...
+<div class="section-example-container">
+<pre class="python">
+    self.AddForex('EURUSD', Resolution.Hour) # Subscribe to hourly QuoteBars    
+ 
+ def OnData(self, data):
+    QuoteBar    = data['EURUSD']
+    BidBar      = QuoteBar.Bid
+    AskBar      = QuoteBar.Ask
+    
+    AskOpen     = AskBar.Open
```
[the remark this answers is not in the document]
Comment (grntmhqo) [reply 02 Algorithm Reference/03 Handling Data/07 QuoteBars.html]:
```suggestion
    AskOpen     = AskBar.Open
```

File: 02 Algorithm Reference/03 Handling Data/07 QuoteBars.html [hunk 13]
```diff
@@ -0,0 +1,42 @@
...
+<p>
+    With a specific QuoteBar you can also access Bid and Ask Bars for the same security. These Bars provide information specific to the Bid and Ask side of Forex and futures secuirites, whereas QuoteBars use the mean of the Open, High, Low, and Close attributes of the Bid and Ask Bars to extrapolte values.
+</p>
+<div class="section-example-container">
+<pre class="python">
+    self.AddForex('EURUSD', Resolution.Hour) # Subscribe to hourly QuoteBars    
+ 
+ def OnData(self, data):
+    QuoteBar    = data['EURUSD']
+    BidBar      = QuoteBar.Bid
```
[the remark this answers is not in the document]
Comment (grntmhqo) [reply 02 Algorithm Reference/03 Handling Data/07 QuoteBars.html]:
```suggestion
    BidBar      = QuoteBar.Bid
```

File: 02 Algorithm Reference/03 Handling Data/07 QuoteBars.html (hunk 7 above)
[the remark this answers is not in the document]
Comment (grntmhqo) [reply 02 Algorithm Reference/03 Handling Data/07 QuoteBars.html]:
```suggestion
    self.AddForex('EURUSD', Resolution.Hour) # Subscribe to hourly QuoteBars
```

File: 02 Algorithm Reference/03 Handling Data/11 Delistings (hunk 5 above)
Comment (grntmhqo) [reply 02 Algorithm Reference/03 Handling Data/11 Delistings]:
```suggestion
    $[Delisting,T:QuantConnect.Data.Market.Delisting] events provide notice an asset is no longer trading on the exchange.
```

**Comment by grntmhqo** [comment]
changes made

File: 02 Algorithm Reference/03 Handling Data/06 TradeBars.html (hunk 6 above)
[the remark this answers is not in the document]
Comment (grntmhqo) [reply 02 Algorithm Reference/03 Handling Data/06 TradeBars.html]: ok i understand the code now but where do i write these commands?in the comments section or in the original
reademe file?and do i send pull request after or push the commit?

File: 02 Algorithm Reference/03 Handling Data/07 QuoteBars.html (hunk 13 above)
[the remark this answers is not in the document]
Comment (AlexCatarino) [reply 02 Algorithm Reference/03 Handling Data/07 QuoteBars.html]: Right. We should use either all lower case with or without underscore(s):
```python
quote_bar = data['EURUSD']
bidbar = quote_bar.Bid
```
but camelCase keeps the examples looking like the C# version.

**Comment by simonsonjack** [comment]
Sorry about all the commit notifications, any in the future will be made locally and then pushed in a chunk. Made stylistic code changes and implemented Jared's suggested changes. Review and let me know if anything else needs to be changed.

**Comment by grntmhqo** [comment]
ok so what steps need to be taken by me to integrate your changes ?

**Pull request merged**